core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala

```diff
@@ -72,8 +75,10 @@ private[deploy] object DeployMessages {
       exception: Option[Exception])
     extends DeployMessage
 
-  case class WorkerSchedulerStateResponse(id: String, executors: List[ExecutorDescription],
-     driverIds: Seq[String])
+  case class WorkerSchedulerStateResponse(
+      id: String,
+      execWithResources: List[(ExecutorDescription, Map[String, Seq[String]])],
```

Comment:
use ResourceAllocation here?   I realize this has extra component info, but @mengxr preferred that be passed even when not necessarily needed to keep things consistent and easier to read.

Reply:
Agreed. I was hesitant about using a less info structure or meaningful but a little bit redundant object previously.